File: frontend/src/types/emissionsSummary.ts
/** Mirrors docs/api-contract.md — Emissions Summary (dashboard) section. */

import type { SourceType } from "./emissionSource";

export interface EmissionsPeriod {
  start: string;
  end: string;
}

export interface EmissionsSummary {
  facility_id: number;
  period: EmissionsPeriod;
  /** Decimal string, e.g. "12045.30". */
  total_emissions_kg_co2e: string;
  by_source_type: Record<SourceType, string>;
}

File: backend/app/schemas/emissions_summary.py
"""Pydantic schema for the facility emissions-summary (dashboard) endpoint.

Matches docs/api-contract.md — Emissions Summary section.
"""

from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class EmissionsPeriod(BaseModel):
    start: date
    end: date


class EmissionsSummaryResponse(BaseModel):
    """GET /facilities/{id}/emissions-summary response."""
    facility_id: int
    period: EmissionsPeriod
    total_emissions_kg_co2e: Decimal
    by_source_type: dict[str, Decimal]
